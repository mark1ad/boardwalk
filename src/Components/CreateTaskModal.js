import React, { Component } from "react";
import { PropTypes } from "prop-types";

class CreateTaskModal extends Component {
  render() {
    // Render nothing if the "show" prop is false
    if (!this.props.show) {
      return null;
    }

    return(
      <div>
        <h1>Hi!</h1>
      </div>
    );
  }
}

CreateTaskModal.propTypes = {
  show: PropTypes.bool.isRequired
};

export default CreateTaskModal;
