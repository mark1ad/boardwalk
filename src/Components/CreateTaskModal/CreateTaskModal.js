import React, { Component } from "react";
import { PropTypes } from "prop-types";

/**
 * Modal form for entering information for a new task
 */
class CreateTaskModal extends Component {
  render() {
    // Render nothing if the "show" prop is false
    if (!this.props.show) {
      return null;
    }

    return(
      <div className="create-task-modal">
        <div className="modal-style">
          <h1>Hi!</h1>
          <h2>Tasklist: {this.props.tasklist_id}</h2>
        </div>
      </div>
    );
  }
}

CreateTaskModal.propTypes = {
  /** If true, modal is displayed */
  show: PropTypes.bool.isRequired,
  /** ID of tasklist that new task will be in. */
  tasklist_id: PropTypes.number
};

export default CreateTaskModal;
