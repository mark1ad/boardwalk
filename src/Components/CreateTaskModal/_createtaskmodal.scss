.create-task-modal {
  position: absolute;
  z-index: 1;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: rgba(0,0,0, 0.3);
  padding: 50px
}

.modal-style {
  background-color: white;
  border-radius: 5px;
  max-width: 500px;
  min-height: 300px;
  margin: 0 auto;
  padding: 30px;
}
