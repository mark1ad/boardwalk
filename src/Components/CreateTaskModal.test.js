import React from "react";
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer';

import CreateTaskModal from "./CreateTaskModal";

it("render without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<CreateTaskModal show={true} />, div);
})

it('renders correctly with show==true', () => {
  const tree = renderer
    .create(<CreateTaskModal show={true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly with show == false', () => {
  const tree = renderer
    .create(<CreateTaskModal show={false} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})
