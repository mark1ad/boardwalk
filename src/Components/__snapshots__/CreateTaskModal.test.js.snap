// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with show == false 1`] = `null`;

exports[`renders correctly with show==true 1`] = `
<div>
  <h1>
    Hi!
  </h1>
</div>
`;
